### ISSUE TITLE
LanguagePrimitives error strings are initialized in default culture before usage in different culture

### ISSUE THREAD

**Issue by latkin** [Microsoft/visualfsharp#159]
Ref #141 

Some localized error strings in [prim-types.fs](https://github.com/microsoft/visualfsharp/blob/fsharp4/src/fsharp/FSharp.Core/prim-types.fs#L660-L669) are declared as module values, which causes them to be statically initialized long before they might be used. This means they might be initialized in a different culture than when they are actually used, which is not desirable.

Fix is to refactor these to be evaluated on-demand.

``` fsharp
    try
        Seq.min []
    with e -> 
        printfn "%A" e //use default currentUICulture of process

    // change culture now
    System.Threading.Thread.CurrentThread.CurrentUICulture <- new CultureInfo("it-IT")

    try
        Seq.min []
    with e -> 
        printfn "%A" e // e message is wrong, because is not of 'it-IT' culture
```

**Comment by latkin** [comment]
Re-opening this one as the fix was rolled back before 4.0

**Comment by forki** [comment]
@latkin do you remember why this was rolled back?

**Comment by latkin** [comment]
@forki 

https://github.com/Microsoft/visualfsharp/issues/486

https://github.com/Microsoft/visualfsharp/pull/497

**Comment by enricosada** [comment]
@forki because i changed the function to be lazy loaded. But that was a breaking change for binary api ( these function are inlined inside the compiled assembly ), so assemblies compiled with old fsharp.core but using newer fsharp.core didnt find the functions => runtime crash.

I fixed this issue with #694 , but is should wait for next fsharp.core major version (5?). That's because code compiler with `4.1` will need `>= 4.1`, and that's not acceptable.
Also the pr has label `committed` **but it's not**. 
All explained there https://github.com/Microsoft/visualfsharp/pull/694#issuecomment-149828712

So let's wait fsharp.core 5.0 :smile:

**Comment by dsyme** [comment]
I'm closing this as Wont Fix.  The problem is in practice very minor, and the original fix was rolled back because it would necessitate a breaking change
